source/agora/consensus/protocol/Nominator.d

```diff
@@ -583,7 +591,8 @@ extern(D):
             if (!proc)
             {
                 this.queued_envelopes.insertBack(envelope.serializeFull.deserializeFull!SCPEnvelope());
-                this.timers[TimersIdx.Envelope].rearm(EnvTaskDelay, false);
+                if (!this.is_shutting_down)
+                    this.timers[TimersIdx.Envelope].rearm(EnvTaskDelay, false);
```

Comment:
There are codes that look like this. So how about creating a private `startTimer` function that could be used for all the timers including the `Nomination` timer?

Reply:
We need to keep the public `startNominatingTimer` function as it is called when the node has been enrolled from `Validator` class. However I can still create a `startTimer` function to reduce the repetition.

Reply:
Right, we do. And I meant that the `startNominatingTimer` calls.

Reply:
ok see updates. I also made same changes to `FullNode` timers.